tripy/examples/resent50/resnet50.ipynb

```diff
@@ -0,0 +1,529 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ResNet50 Image Classificiation with `tripy` - An Implementation and Demo\n",
+    "\n",
+    "This notebook aims to demonstrate the implementation of a ResNet50 model using `tripy`. We'll explore the architecture, \n",
+    "load pretrained weights, and run predictions on a sample dataset, showcasing how `tripy` can be used effectively for inference of ResNet50 based image classification model.\n",
+    "\n",
+    "### Objectives:\n",
+    "1. Implement ResNet50 using `tripy` and load pretrained weights.\n",
+    "2. Run predictions on sample images and visualize results.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Setup\n",
+    "\n",
+    "install neccessary libraries, load pretrained resnet50 weights and the inference dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install datasets matplotlib pillow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load necessary libraries\n",
+    "import torch\n",
+    "from transformers import ResNetForImageClassification, AutoImageProcessor\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "# Load the pretrained ResNet50 model from Hugging Face\n",
+    "resnet_pretrained = ResNetForImageClassification.from_pretrained(\"microsoft/resnet-50\")\n",
+    "resnet_pretrained.eval()\n",
+    "resnet_pretrained = resnet_pretrained.to('cuda')\n",
+    "\n",
+    "# Load the image processor\n",
+    "processor = AutoImageProcessor.from_pretrained(\"microsoft/resnet-50\")\n",
+    "\n",
+    "# Load a sample image from the dataset\n",
+    "dataset = load_dataset(\"comet-team/coco-500\", split=\"train\")\n",
+    "idx = 103\n",
+    "image = dataset[idx]['Image']\n",
+    "image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Preprocess the image\n",
+    "inputs = processor(image, return_tensors=\"pt\")['pixel_values'].to('cuda')\n",
+    "\n",
+    "# Run the image through the model\n",
+    "with torch.no_grad():\n",
+    "    logits = resnet_pretrained(inputs).logits\n",
+    "\n",
+    "# Get the predicted label\n",
+    "predicted_label = logits.argmax(-1).item()\n",
+    "print(f\"Predicted Label: {resnet_pretrained.config.id2label[predicted_label]}\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tripy Implementation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tripy as tp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will define:\n",
+    "\n",
+    "1. **ResNet Building Blocks**:\n",
```

Comment:
Maybe we can include this information as docstrings/comments in the modules themselves? I think it would be helpful to have this high-level information where we can also look at the code.

Reply:
yeah will include it as well.